src/main/java/org/broadinstitute/hellbender/tools/exome/allelefraction/AlleleFractionLikelihoods.java

```diff
@@ -0,0 +1,179 @@
+     * We need such functionality because MCMC evaluates the likelihood under proposed parameter changes.
+     *
+     * @param state allele fraction state
+     * @param segment index of segment containijng this het site
+     * @param count AllelicCount of alt and ref reads
+     * @param indicator the hidden state (alt minor / ref minor / outlier)
+     *
+     * @return if indicator == ALT_MINOR:
+     * <p>
+     * log { [beta^alpha / Gamma(alpha)][(1-pi)/2] * int_{0 to infty} f^a * (1-f)^r * lambda^(alpha + r - 1) * exp(-beta*lambda)/(f + (1-f)*lambda)^n d lambda }
+     * <p>
+     * if indicator == REF_MINOR, same as ALT_MINOR but with f <--> 1 - f
+     * <p>
+     * if indicator == OUTLIER log {pi * a!r!/(n+1)!}
+     * <p>
+     * where alpha = mu*beta and n = a + r
+     */
+    public static double hetLogLikelihood(final AlleleFractionState state, final int segment, final AllelicCount count, final AlleleFractionIndicator indicator) {
+        return hetLogLikelihood(state, segment, count, indicator, AllelicPanelOfNormals.EMPTY_PON);
+    }
+
+    public static double hetLogLikelihood(final AlleleFractionState state, final int segment, final AllelicCount count, final AlleleFractionIndicator indicator,
+                                          final AllelicPanelOfNormals allelicPON) {
+        final SimpleInterval site = count.getInterval();
+        final double alpha;
+        final double beta;
+        if (allelicPON.equals(AllelicPanelOfNormals.EMPTY_PON)) {
+            beta = state.meanBias() / state.biasVariance();
+            alpha = state.meanBias() * beta;
+
+        } else {
+            alpha = allelicPON.getAlpha(site);
+            beta = allelicPON.getBeta(site);
+        }
+        final double pi = state.outlierProbability();
+        final double minorFraction = state.minorFractionInSegment(segment);
+        final int a = count.getAltReadCount();
+        final int r = count.getRefReadCount();
+
+        return hetLogLikelihood(alpha, beta, minorFraction, pi, indicator, a, r);
+    }
+
+    public static double hetLogLikelihood(final double alpha, final double beta, final double minorFraction,
+                                          final double outlierProbability, final AlleleFractionIndicator indicator,
+                                          final int a, final int r) {
+        if (indicator == AlleleFractionIndicator.OUTLIER) {
+            return log(outlierProbability) + log10ToLog(log10Factorial(a) + log10Factorial(r) - log10Factorial(a + r + 1));
+        } else {
+            final double f = indicator == AlleleFractionIndicator.ALT_MINOR ? minorFraction : 1 - minorFraction;
+            return log((1 - outlierProbability) / 2) + logPhi(alpha, beta, f, a, r);
+        }
+    }
+
+    protected static double logPhi(final double alpha, final double beta, final double f, final int a, final int r) {
```

Comment:
In light of comments below, this could be renamed to something like `logIntegralOverAllelicBiasRatio` or something.